### PR TITLE
Release: 3.0.0

### PR DESCRIPTION
**Pull request by mashapedeployment**

### CODE REVIEW

**Comment by curiositycasualty** [comment]
Drafting while I resolve the inclusion of the atc-router.

File: Formula/kong.rb [hunk 1]
```diff
@@ -92,3 +94,53 @@ def install
...
+diff -r -u a/kong/cmd/prepare.lua b/kong/cmd/prepare.lua
+--- a/kong/cmd/prepare.lua	2022-09-12 14:38:55.000000000 +0200
++++ b/kong/cmd/prepare.lua	2022-09-15 10:53:58.000000000 +0200
+@@ -23,8 +23,8 @@
+
+ Example usage:
+  kong migrations up
+- kong prepare -p /usr/local/kong -c kong.conf
+- nginx -p /usr/local/kong -c /usr/local/kong/nginx.conf
++ kong prepare -p HOMEBREW_PREFIX -c kong.conf
```
Comment (kikito) [inline Formula/kong.rb:106]: I am not fully understanding how HOMEBREW_PREFIX is working on this diff. If this is an environment variable, shouldn't it be preceded by `$` ?

```suggestion
+ kong prepare -p $HOMEBREW_PREFIX -c kong.conf
```

Perhaps homebrew is doing some magic on the diffs that I am missing.

File: Formula/kong.rb [hunk 2]
```diff
@@ -92,3 +94,53 @@ def install
...
+
+diff -r -u a/kong/runloop/plugin_servers/process.lua b/kong/runloop/plugin_servers/process.lua
+--- a/kong/runloop/plugin_servers/process.lua	2022-09-12 14:38:55.000000000 +0200
++++ b/kong/runloop/plugin_servers/process.lua	2022-09-15 10:54:10.000000000 +0200
+@@ -61,7 +61,7 @@
+       local env_prefix = "pluginserver_" .. name:gsub("-", "_")
+       _servers[i] = {
+         name = name,
+-        socket = config[env_prefix .. "_socket"] or "/usr/local/kong/" .. name .. ".socket",
++        socket = config[env_prefix .. "_socket"] or "HOMEBREW_PREFIX/" .. name .. ".socket",
```
Comment (kikito) [inline Formula/kong.rb:131]: Isn't this code trying to open a socket inside a folder called `HOMEBREW_PREFIX/` (if the config doesn't have a value for `config[env_prefix .. "_socket"]`)?

File: Formula/kong.rb (hunk 1 above)
Comment (kikito) [reply Formula/kong.rb:106]: Found it in [the docs](https://docs.brew.sh/Formula-Cookbook#patches):

> In embedded patches, the string “HOMEBREW_PREFIX” is replaced with the value of the constant HOMEBREW_PREFIX before the patch is applied.

File: Formula/openresty@1.21.4.1.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    # https://forums.developer.apple.com/thread/121887
+    ENV.append_to_cflags "-fno-stack-check" if DevelopmentTools.clang_build_version >= 1010
+
+    args = [
+      "--prefix #{prefix}",
+      "--openresty #{OPENRESTY_VERSION}",
+      "--openssl #{OPENSSL_VERSION}",
+      "--luarocks #{LUAROCKS_VERSION}",
+      "--pcre #{PCRE_VERSION}",
+      "-j #{ENV.make_jobs}",
```
Comment (gszr) [inline Formula/openresty@1.21.4.1.rb:39]: I think we also need to specify new deps like `--atc-router`, `--resty-events`, `--resty-lmdb`, `--resty-websocket`

**Review by kikito — APPROVED** [review]

**Comment by gszr** [comment]
@kikito I don't think this was ready to merge yet.
> I think we also need to specify new deps like --atc-router, --resty-events, --resty-lmdb, --resty-websocket
Still stands; plus, @curiositycasualty is still working on the opentelemetry proto files.

**Comment by hanshuebner** [comment]
> @curiositycasualty is still working on the opentelemetry proto files.

The proto files are in the `include/` directory which is now copied into the Cellar.  The Formula works for me.  Does it not work for you?

File: Formula/openresty@1.21.4.1.rb (hunk 3 above)
Comment (kikito) [reply Formula/openresty@1.21.4.1.rb:46]: I answered to this message but it seems my answer was lost (or I forgot to press "send"). Anyway, yes, I added the flags as you requested, thank you for pointing those were missing.

**Pull request merged**